src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSparkOptimized.java

```diff
@@ -0,0 +1,205 @@
+            // TODO: fix
+            logger.info("(HACK): copying the GCS file to local just so we can read it back.");
+            hacking.start();
+            localVariants = new ArrayList<>();
+            // this only works with the API_KEY, but then again it's a hack so there's no point in polishing it.
+            PipelineOptions popts = auth.asPipelineOptionsOldStyle();
+            int i=0;
+            for (String s : knownVariants) {
+                // presumably they're all GCS.
+                String d = IOUtils.createTempFile("knownVariants-"+i,".vcf").getAbsolutePath();
+                try {
+                    BucketUtils.copyFile(s, popts, d);
+                } catch (IOException x) {
+                    throw new UserException.CouldNotReadInputFile(s,x);
+                }
+                i++;
+                localVariants.add(d);
+            }
+            hacking.stop();
+            logger.info("Copying the vcf took "+hacking.elapsed(TimeUnit.MILLISECONDS)+" ms.");
+        }
+
+        List<Variant> variants = VariantsSource.getVariantsList(localVariants);
+
+        List<SimpleInterval> shardedIntervals = IntervalUtils.cutToShards(intervals, bigShardSize);
+        ArrayList<ContextShard> localShards = AddContextDataToReadOptimizedFn.fillVariants(shardedIntervals, variants, margin);
+
+        logger.info("Shipping "+localShards.size()+" big shards.");
+        Stopwatch shipping = Stopwatch.createStarted();
+        JavaRDD<ContextShard> shards = ctx.parallelize(localShards);
+        System.out.println("Shipping took: "+shipping.elapsed(TimeUnit.MILLISECONDS) + " ms.");
+
+        logger.info("Loading reads");
+        JavaRDD<ContextShard> reads;
+        try {
+            reads = shards.flatMap( AddContextDataToReadOptimizedFn.subdivideAndFillReads(bam, auth, smallShardSize, margin, readFilterToApply) );
+        } catch (IOException x) {
+            throw new GATKException("Couldn't read "+bam+": "+x.getMessage(), x);
+        }
+
+        // fill in reference bases
+        reads = reads.map(s -> AddContextDataToReadOptimizedFn.fillContext(rds, s));
+
+        logger.info("computing recal tables");
+
+        // run BaseRecalibratorEngine.
+        BaseRecalibratorEngineSpark recal = new BaseRecalibratorEngineSpark(readsHeaderView, refDictionaryView, bqsrArgs);
+        JavaRDD<RecalibrationTables> tables = reads.mapPartitions(s->recal.apply(s));
+
+        logger.info("merging recal tables");
+
+        RecalibrationTables table = tables.reduce( (a,b)->a.combine(b) );
+
+        BaseRecalibrationEngine.finalizeRecalibrationTables(table);
+
+        logger.info("saving final report");
+
+        try {
+            BaseRecalibratorEngineSpark.saveTextualReport(outputTablesPath, readsHeader, table, bqsrArgs, auth);
+        } catch (IOException e) {
+            throw new UserException.CouldNotCreateOutputFile(new File(outputTablesPath), e);
+        }
+
+        System.out.println("Done.");
```

Comment:
I removed it instead, it's not really necessary anyways. Same for the earlier log messages.